go/vt/vtctl/reparentutil/replication.go

```diff
@@ -168,21 +170,27 @@ func StopReplicationAndBuildStatusMaps(
 		statusMap        = map[string]*replicationdatapb.StopReplicationStatus{}
 		primaryStatusMap = map[string]*replicationdatapb.PrimaryStatus{}
 		m                sync.Mutex
-		errChan          = make(chan error)
+		errChan          = make(chan concurrency.Error)
 	)
 
 	groupCtx, groupCancel := context.WithTimeout(ctx, waitReplicasTimeout)
 	defer groupCancel()
 
-	fillStatus := func(alias string, tabletInfo *topo.TabletInfo) {
-		err := vterrors.Errorf(vtrpc.Code_UNAVAILABLE, "fillStatus did not successfully complete")
-		defer func() { errChan <- err }()
+	fillStatus := func(alias string, tabletInfo *topo.TabletInfo, mustWaitForTablet bool) {
+		var concurrencyErr concurrency.Error
+		var err error
+		defer func() {
+			concurrencyErr.Err = err
+			concurrencyErr.MustWaitFor = mustWaitForTablet
+			errChan <- concurrencyErr
+		}()
 
 		logger.Infof("getting replication position from %v", alias)
 
 		_, stopReplicationStatus, err := tmc.StopReplicationAndGetStatus(groupCtx, tabletInfo.Tablet, replicationdatapb.StopReplicationMode_IOTHREADONLY)
-		switch err {
-		case mysql.ErrNotReplica:
+		sqlErr, isSQLErr := mysql.NewSQLErrorFromError(err).(*mysql.SQLError)
+		switch {
+		case isSQLErr && sqlErr != nil && sqlErr.Number() == mysql.ERNotReplica:
 			var primaryStatus *replicationdatapb.PrimaryStatus
```

Comment:
I think it will be more readable to rewrite this switch..case as if..else instead.
```
if err != nil {
  // sql error creation and handling
  if isSQLErr && .... {
     ...
  } else {
    // default case goes here
  }
} else {
// nil error case goes here
}
```
WDYT?

Reply:
I am fine it being either way. If you think this is more readable, I am making this change

Reply:
changed